cli/internal/helm/action.go

```diff
@@ -159,3 +176,84 @@ func (a *upgradeAction) ReleaseName() string {
 func (a *upgradeAction) IsAtomic() bool {
 	return a.helmAction.Atomic
 }
+
+func saveChart(release Release, actionDir string, fileHandler file.Handler) error {
+	if err := saveChartToDisk(release.Chart, actionDir, fileHandler); err != nil {
+		return fmt.Errorf("saving chart %s to %q: %w", release.ReleaseName, actionDir, err)
+	}
+	if err := fileHandler.WriteYAML(filepath.Join(actionDir, release.Chart.Metadata.Name, "overrides.yaml"), release.Values); err != nil {
+		return fmt.Errorf("saving override values for chart %s to %q: %w", release.ReleaseName, actionDir, err)
+	}
+
+	return nil
+}
+
+// saveChartToDisk saves a chart as files in a directory.
+//
+// This takes the chart name, and creates a new subdirectory inside of the given dest
+// directory, writing the chart's contents to that subdirectory.
+// Dependencies are written using the same format, instead of writing them as tar files
+//
+// View the SaveDir implementation in chartutil as reference: https://github.com/helm/helm/blob/3a31588ad33fe3b89af5a2a54ee1d25bfe6eaa5e/pkg/chartutil/save.go#L40
+func saveChartToDisk(c *chart.Chart, dest string, fileHandler file.Handler) error {
```

Comment:
What speaks against using the SDK's `Save` or `SaveDir` functions? I understand it's an extra step for the user to unpack the charts. But they only need to do that to modify the charts. The helm cli supports loading packaged charts. 

Would be less maintenance burden for us if we did use the SDK directly.

Reply:
I would personally prefer writing the files directly, instead of having tar files of the charts. (The code for `SaveDir` also hasn't been touched in the last 4 years, so I think it is unlikely there will be any relevant changes in the future)

Reply:
Even if it didn't change I would prefer to use the SDK's interface. Let's let @malt3 decide.

Reply:
I also agree with @derpsteb. What is the disadvantage of storing tars? Afais, the files will only be checked in rare cases anyways?

Reply:
My reasoning for saving the files instead of tar was that it is way easier to look at the chart that way.
Since @malt3 was the one who brought this feature up, I would like to hear his opinion on this.
If we are OK with using tars, I would opt to just use [`chartutil.Save`](https://pkg.go.dev/helm.sh/helm/v3/pkg/chartutil#Save) which saves everything as tar, not just the dependencies.

Reply:
One reason I want to store the helm charts to disk is that I want to be able to modify the helm values (if necessary) because they are no longer usable. The example we saw in a user deployment was a reference to an old Azure service account that had to be replaced due to a customer-internal IAM change.
So making these easily editable (and storing yaml files directly) is preferable.

Reply:
The values are stored separately tho (at least our overrides), they are not saved by the helm `chartutil` code.

Reply:
Oh that makes sense. In that case my preference is still slightly towards being able to read/edit the helm charts easily since in very rare cases, a customer may actually need to edit the charts.